### PR TITLE
feat(metrics_extraction): Support extracting multiple spec versions

### PR DESCRIPTION
**Pull request by armenzg**
Adding new specification extraction versions is as simple as adding a new configuration to `SPEC_VERSIONS` and accessing `self.spec_version` within `OnDemandMetricSpec`.

We can use `MIN_VERSION` as a means of declaring which versions are still extracting.

### CODE REVIEW

File: src/sentry/relay/config/metric_extraction.py [hunk 1]
```diff
@@ -144,35 +145,12 @@ def _get_alert_metric_specs(
                 tags={"prefilling": prefilling, "dataset": alert_snuba_query.dataset},
             )
 
-            if result := _convert_snuba_query_to_metric(
-                project, alert_snuba_query, prefilling, use_updated_env_logic=True
-            ):
-                _log_on_demand_metric_spec(
```
Comment (armenzg) [inline src/sentry/relay/config/metric_extraction.py:150]: There's a right shift for this block.

File: src/sentry/relay/config/metric_extraction.py [hunk 2]
```diff
@@ -509,26 +489,29 @@ def _convert_aggregate_and_query_to_metric(
...
-        # TODO: switch to validate_rule_condition
-        if (condition := metric_spec.get("condition")) is not None:
-            validate_sampling_condition(json.dumps(condition))
-        else:
-            metrics.incr(
-                "on_demand_metrics.missing_condition_spec",
-                tags={"prefilling": prefilling},
+        metric_specs_and_hashes = []
+        # Create as many specs as we support
+        for spec_version in get_spec_versions():
```
Comment (armenzg) [inline src/sentry/relay/config/metric_extraction.py:494]: We now iterate over all active versions.

File: src/sentry/snuba/metrics/extraction.py [hunk 3]
```diff
@@ -49,6 +50,41 @@
...
+# Once we're ready to abandon a version bump this value
+MIN_VERSION = 0
+
+SPEC_VERSIONS = [
+    SpecVersion(0),
+    SpecVersion(1, ["use_updated_env_logic"]),
+]
+
+
+def get_spec_versions(min_version: int = 0) -> Sequence[SpecVersion]:
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/extraction.py:74]: Can we make an explicit constant for `min_version`? To make it clear

**Review by iambriccardo — APPROVED** [review]

**Comment by iambriccardo** [comment]
🚢

**Pull request merged**